src/pipelinedefinition.cpp

```diff
@@ -384,119 +387,149 @@ class NodeValidator {
         }
 
         // If dependency node is of type DL model, make sure there is underlying model output present.
-        if (dependencyNodeInfo.kind == NodeKind::DL) {
+        if (dependencyNodeInfo.kind == NodeKind::DL || dependencyNodeInfo.kind == NodeKind::CUSTOM) {
             // Check whether underlying model contains required output.
             const auto& modelOutputName = dependencyNodeInfo.outputNameAliases.at(dataSource);
-            if (dependencyModelInstance->getOutputsInfo().count(modelOutputName) == 0) {
-                SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline({}) definition failed. Missing model (name:{}, version:{}) output:{} of dependency node:{}",
+            if (this->dependencyOutputsInfo.count(modelOutputName) == 0) {
+                SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline({}) definition failed. Missing output:{} of dependency node:{}; data source {}",
                     pipelineName,
-                    dependencyNodeInfo.modelName,
-                    dependencyNodeInfo.modelVersion.value_or(0),
                     modelOutputName,
-                    dependencyNodeInfo.nodeName);
+                    dependencyNodeInfo.nodeName,
+                    dataSource);
                 return StatusCode::PIPELINE_NODE_REFERING_TO_MISSING_MODEL_OUTPUT;
             }
         }
 
         return StatusCode::OK;
     }
 
-    Status checkConnectionMetadataCorrectness(const NodeInfo& dependencyNodeInfo, std::shared_ptr<ModelInstance>& dependencyModelInstance, const std::string& modelInputName, const std::string& modelOutputName) {
-        // If validated connection pair connects two DL model nodes,
+    Status influenceShapeWithDemultiplexer(shape_t& shape, const NodeInfo& demultiplicatorNodeInfo) {
+        if (shape.size() < 3) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline({}) definition failed. Node {} demultiply cannot occur due to not enough shape dimensions: {}",
+                this->pipelineName,
+                demultiplicatorNodeInfo.nodeName,
+                shape.size());
+            return StatusCode::PIPELINE_NOT_ENOUGH_SHAPE_DIMENSIONS_TO_DEMULTIPLY;
+        }
+        if (shape[1] != demultiplicatorNodeInfo.demultiplyCount.value()) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline({}) definition failed. Demultiply count: {} of node: {} does not match tensor second dimenson value: {}",
```

Comment:
The same as in L451: `Validation of pipeline: {}`

Reply:
Done